nixos/modules/services/web-apps/porn-vault/default.nix

```diff
@@ -0,0 +1,115 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  cfg = config.services.porn-vault;
+  configFormat = pkgs.formats.json { };
+  defaultConfig = import ./default-config.nix;
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    getExe
+    literalExpression
+    types
+    ;
+in
+{
+  options = {
+    services.porn-vault = {
+      enable = lib.mkEnableOption "Porn-Vault";
+
+      package = lib.mkPackageOption pkgs "porn-vault" { };
+
+      autoStart = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Whether to start porn-vault automatically.
+        '';
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 3000;
+        description = ''
+          Which port Porn-Vault will use.
+        '';
+      };
+
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Whether to open the Porn-Vault port in the firewall.
+        '';
+      };
+
+      settings = {
+        enable = mkEnableOption "Porn-Vault configuration";
+        default = true;
+
+        settings = mkOption {
+          type = configFormat.type;
+          description = ''
+            Configuration for Porn-Vault. The attributes are serialized to JSON in config.json.
+
+            See https://gitlab.com/porn-vault/porn-vault/-/blob/dev/config.example.json
+          '';
+          default = defaultConfig;
+          apply = lib.recursiveUpdate defaultConfig;
+        };
+      };
```

Comment:
```suggestion
      settings = mkOption {
        type = configFormat.type;
        description = ''
          Configuration for Porn-Vault. The attributes are serialized to JSON in config.json.

          See https://gitlab.com/porn-vault/porn-vault/-/blob/dev/config.example.json
        '';
        default = defaultConfig;
        apply = lib.recursiveUpdate defaultConfig;
      };
```

Reply:
I believe it's nice to let people disable the nix-managed settings if they want tbh. I've just force-pushed a fix that makes settings.enable true by default tho!